### PR TITLE
Update lodash

### PR DESCRIPTION
**Pull request by xPaw**
Hopefully I didn't miss any function changes.

### CODE REVIEW

**Comment by astorije** [comment]
Let's have 2 :+1: on this one, it's quite a dangerous PR and needs to be tested fairly well :-)
Thanks a lot for this @xPaw!

**Comment by astorije** [comment]
Can @JocelynDelalande, @YaManicKill or @wizardfrag self-assign this one please? :-)
(I guess that's where organization teams are interesting :p)

**Comment by JocelynDelalande** [comment]
As I'm not myself doogfooding shout, I'm not useful for that kind of review that requires testing in real conditions, for some time, using all features to see where it may crashe.

File: src/clientManager.js [hunk 1]
```diff
@@ -135,4 +135,3 @@ ClientManager.prototype.autoload = function(/* sockets */) {
 		});
 	}, 1000);
 };
-
```
Comment (astorije) [inline src/clientManager.js]: Was that intentional? We should always keep an EOF line to keep git and other tools happy :-)

File: src/clientManager.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ ClientManager.prototype.removeUser = function(name) {
 ClientManager.prototype.autoload = function(/* sockets */) {
 	var self = this;
 	setInterval(function() {
-		var loaded = _.pluck(
+		var loaded = _.map(
 			self.clients,
```
Comment (astorije) [inline src/clientManager.js]: Unless I am being mistaken (it's late :-)), it seems like this is the same as:

```js
self.clients.map(function (client) {
  return client.name;
});
```

Example:

```js
> var arr = [{foo: 1}, {foo: 2}, {foo: 3}];
> _.map(arr, "foo");
[ 1, 2, 3 ]
> arr.map(function (x) { return x.foo; });
[ 1, 2, 3 ]
```

While I understand you are merely upgrading as-is here, I believe we should leverage vanilla JS whenever possible, and this is a good case :-)
What do you think?

File: src/models/network.js [hunk 3]
```diff
@@ -46,8 +46,8 @@ Network.prototype.export = function() {
 		"commands"
 	]);
 	network.nick = (this.irc || {}).me;
-	network.join = _.pluck(
-		_.where(this.channels, {type: "channel"}),
+	network.join = _.map(
+		_.filter(this.channels, {type: "channel"}),
```
Comment (astorije) [inline src/models/network.js]: Same as above, I'd rather see:

```js
this.channels.filter(function (channel) {
  return channel.type === "channel";
}).map(function (channel) {
  return channel.name;
}).join(",");
```

Yes, it's a few extra chars, but this gets us as close as the language as possible so I think it's a win! Plus we can even improve it with `.reduce()` if necessary (but I don't believe it is)!

File: src/plugins/irc-events/kick.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ module.exports = function(irc, network) {
 	var client = this;
 	irc.on("kick", function(data) {
 		var from = data.nick;
-		var chan = _.findWhere(network.channels, {name: data.channel});
+		var chan = _.find(network.channels, {name: data.channel});
```
Comment (astorije) [inline src/plugins/irc-events/kick.js]: I'm sure there is an ES5 equivalent here too, but I won't be as picky a with `.map()` and `.filter()` which are now well accepted.

**Comment by astorije** [comment]
(Taking ownership and first review then)

So far so goo, @xPaw, but I left you a few comments inline :-)

Also, just wondering, did you try something like https://github.com/lodash/lodash-migrate to migrate? Not that we should blindly rely on that, but that might give you some extra confidence that nothing was forgotten.

Thanks!

File: src/clientManager.js (hunk 1 above)
Comment (xPaw) [inline src/clientManager.js]: It had two new lines at the end, my editor automatically removed the extra one.

File: src/clientManager.js (hunk 2 above)
Comment (xPaw) [inline src/clientManager.js]: Not a bad idea, however I would get the upgrade out of the way, and then look into using vanillajs.

**Comment by xPaw** [comment]
> Also, just wondering, did you try something like https://github.com/lodash/lodash-migrate to migrate?

I did just now, it didn't seem to complain.

**Comment by astorije** [comment]
:+1:

**Comment by YaManicKill** [comment]
Cool, looks good to me. 

:+1: Merging.

**Pull request merged**